airbyte-integrations/connectors/destination-bigquery/src/test-integration/java/io/airbyte/integrations/destination/bigquery/BigQueryDestinationTest.java

```diff
@@ -108,81 +111,149 @@ class BigQueryDestinationTest {
+  protected static Map<String, JsonNode> configs;
+  protected static ConfiguredAirbyteCatalog catalog;
+  protected boolean bqTornDown = false;
+  protected boolean gcsTornDown = false;
+
+  private AmazonS3 s3Client;
 
-  private static Stream<Arguments> datasetIdResetterProvider() {
-    // parameterized test with two dataset-id patterns: `dataset_id` and `project-id:dataset_id`
+  private Stream<Arguments> successTestConfigProvider() {
     return Stream.of(
-        Arguments.arguments(new DatasetIdResetter(config -> {})),
-        Arguments.arguments(new DatasetIdResetter(
-            config -> {
-              final String projectId = config.get(BigQueryConsts.CONFIG_PROJECT_ID).asText();
-              final String datasetId = config.get(BigQueryConsts.CONFIG_DATASET_ID).asText();
-              ((ObjectNode) config).put(BigQueryConsts.CONFIG_DATASET_ID,
-                  String.format("%s:%s", projectId, datasetId));
-            })));
+        Arguments.of("config"),
+        Arguments.of("configWithProjectId"),
+        Arguments.of("gcsStagingConfig")
+    );
   }
 
-  @BeforeEach
-  void setup(final TestInfo info) throws IOException {
-    if (info.getDisplayName().equals("testSpec()")) {
-      return;
-    }
+  private Stream<Arguments> failCheckTestConfigProvider() {
+    return Stream.of(
+        Arguments.of("configWithBadProjectId", "User does not have bigquery.datasets.create permission in project"),
+        Arguments.of("insufficientRoleConfig", "User does not have bigquery.datasets.create permission"),
+        Arguments.of("nonBillableConfig", "Access Denied: BigQuery BigQuery: Streaming insert is not allowed in the free tier")
+    );
+  }
+
+  private Stream<Arguments> failWriteTestConfigProvider() {
+    return Stream.of(
+        Arguments.of("configWithBadProjectId", "User does not have bigquery.datasets.create permission in project"),
+        Arguments.of("insufficientRoleConfig", "Permission bigquery.tables.create denied")
+    );
+  }
 
-    if (!Files.exists(CREDENTIALS_PATH)) {
+  @BeforeAll
+  public static void beforeAll() throws IOException {
+    if (!Files.exists(CREDENTIALS_STANDARD_INSERT_PATH)) {
       throw new IllegalStateException(
-          "Must provide path to a big query credentials file. By default {module-root}/config/credentials.json. Override by setting setting path with the CREDENTIALS_PATH constant.");
+          "Must provide path to a big query credentials file. By default destination-bigquery/secrets/credentials-standard.json");
+    }
+    if (!Files.exists(CREDENTIALS_WITH_MISSED_CREATE_DATASET_ROLE_PATH)) {
+      throw new IllegalStateException("""
+                                      Json config not found. Must provide path to a big query credentials file,
+                                       please add file with creds to
+                                      ../destination-bigquery/secrets/credentials-with-missed-dataset-creation-role.json.""");
```

Comment:
```suggestion
                                      <...>/destination-bigquery/secrets/credentials-with-missed-dataset-creation-role.json.""");
```
since `..` might get misinterpreted